File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    
    var bin = n.toString(2)
    bin = bin.replaceAll("0","")
    var len = bin.length
    var answer = 0
    
    while(true){
        n++
        var next = n.toString(2)
        next = next.replaceAll("0","")
        var nextLen = next.length
        if(len == nextLen){
            answer = n
            break
        }
    }
    
    return answer;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        
        String bin = Integer.toBinaryString(n);
        bin = bin.replace("0","");
        int len = bin.length();
        int answer = 0;
        
        while(true){
            n++;
            String next = Integer.toBinaryString(n);
            next = next.replace("0","");
            int nextLen = next.length();
            if(len == nextLen) {
                answer = n;
                break;
            }           
        }
        
        return answer;
    }
}